src/main/java/org/graylog2/gelfclient/transport/AbstractGelfTransport.java

```diff
@@ -39,6 +39,7 @@
     protected final BlockingQueue<GelfMessage> queue;
 
     private final EventLoopGroup workerGroup;
+    GelfSenderThread senderThread;
```

Comment:
Since this field cannot be `final`, I wonder if this class is still thread safe. If someone calls `createBootstrap()` from multiple threads concurrently, there will be a race between the threads to set this field.

What do you think? We might have to lock access to this field or use a `AtomicReference<GelfSenderThread>` to avoid concurrency issues.

Reply:
Good catch. I have moved the initialization of the `senderThread` field to the AbstractGelfTransport constructor and finalized the field in commit https://github.com/Graylog2/gelfclient/pull/43/commits/7d2b04bf8acdb666a4aae48b7439645fbba16a08 I think this should resolve the concern.

Reply:
Yes, way better than locking! :+1:

Reply:
Although: Does reconnecting still works as before? On reconnect we call `createBootstrap()`. In the old version we created a new `GelfSenderThread` on every `createBootstrap()` call. With this change we only create a single object and never create a new one.

Reply:
I looked at this, and the only way I see to solve this is to use an `AtomicReference`, and initialize that in the constructor (like you initially said). The actual `GelfSenderThread` can then still be set in the `createBootstrap` method as it was before. I made a pass at this in https://github.com/Graylog2/gelfclient/pull/43/commits/785ffab3181f2737a35340be50602b89e86fd138. Any other option that I looked at would require bigger interface changes, which are probably not good. Does the reference to `senderThread` look thread safe to you now? Reconnect tests successfully for me with this.

I found another issue where reconnects are still attempted when the `transport.stop()` method is called (and when my new `flushAndStopSynchronously` method is called). I'll create a separate issue for this since it's unrelated.